### DISCUSSION TITLE
Can failureWorkflow be dynamically defined?

### DISCUSSION THREAD

**Discussion by rickyschechter** [Netflix/conductor#2489] [Q&A] [answered]
I see the value is a String and I would assume that it is ONLY the workflow name, so it must be pre-defined in metadata and cannot be defined on the fly, is that correct?

**Comment by kishorebanala** [comment, marked as answer]
@rickyschechter Correct, failureWorkflow is pre-defined and doesn't support dynamic definitions. Please feel free to open a proposal if that's something that would help your use case.

**Reply by rickyschechter** [reply]
Thank you for your response.
Yes, due to multiple reasons, I need to define the workflow dynamically and a static (metadata pre-defined) failureWorkflow is insufficient. 
Having assumed this is the case, in the meantime I have implemented (not happy about it) a poll mechanism where my system checks constantly for status out of synchronization by getting the workflow's status and if different (only expected to be in a failure situation) it adjusts my object. Ideally I have the failureWorkflow update my object with an http task or alike.
What I need is the ability to include the WokflowDef block in such tag. 

What are the steps to properly open a proposal and what information does it need to include?

Thanks!

**Reply by kishorebanala** [reply]
@rickyschechter One option you could consider is to use dynamic workflow definition when starting a workflow, and providing this failure workflow at workflow start time.

> What are the steps to properly open a proposal and what information does it need to include?

Please open a discussion with brief proposal of the problem and possible solutions. Once we reach consensus, we can work on the next steps. Thank you!

**Reply by rickyschechter** [reply]
_"One option you could consider is to use dynamic workflow definition when starting a workflow, and providing this failure workflow at workflow start time."_

Could you please elaborate a little bit? how do I specify this extra definition in my dynamic workflow? This would be an on the fly independent workflow definition that I do not want to be executed but referenceable by failureWorkflow by its name. That would work, if I can have it upload it into the system together with my main workflow definition.

Please advise, thank you!

**Reply by kishorebanala** [reply]
@rickyschechter I was referring to this: https://netflix.github.io/conductor/gettingstarted/startworkflow/#dynamic-workflows, which allows you to provide a workflow definition on the go, in which you should be able to include failureWorkflow as you seem fit. You cannot probably change the failureWorkflow after workflow start though.

**Reply by rickyschechter** [reply]
Right, I am using dynamic workflows (for my main workflow) where for the associated failureWorkflow defined inside, I could ONLY state its name, that I did accomplish it by creating it on the fly (when needed) through a metadata definition prior to submitting the dynamic. It work, but not too clean unfortunately.
Thanks!

**Comment by rickyschechter** [comment]
@kishorebanala I am asking as I believe you are not suggesting that such workflow failure be pre-defined in metadata correct?
